docs/ide/mcp.md

```diff
@@ -0,0 +1,166 @@
+    + You can use [GitHub Copilot for Free](copilot-free-plan.md). Sign up and leverage AI to code faster and more efficiently.
++ Enable [agent mode in GitHub Copilot Chat](copilot-agent.md).
+
+## How do MCP and Visual Studio extend GitHub Copilot's agent?
+
+* MCP clients (like Visual Studio) connect to MCP servers and request actions on behalf of the AI model
+* MCP servers provide one or more tools that expose specific functionalities through a well-defined interface.
+* The Model Context Protocol (MCP) defines the message format for communication between clients and servers, including tool discovery, invocation, and response handling
+
+For example, a file system MCP server might provide tools for reading, writing, or searching files and directories. GitHub's MCP server offers tools to list repositories, create pull requests, or manage issues. MCP servers can run locally on your machine or be hosted remotely, and Visual Studio supports both configurations.
+
+By standardizing this interaction, MCP eliminates the need for custom integrations between each AI model and each tool. This allows you to extend your AI assistant's capabilities by simply adding new MCP servers to your workspace. Learn more about the Model Context Protocol specification.
+
+# Quick Start
+## Configuration example with GitHub MCP server
+
+1. Create a new file: `<SOLUTIONDIR>\.vs\mcp.json`
+2. Paste the following contents into the `mcp.json` file
+
+```markdown
+{
+  "inputs": [
+    {
+      "type": "promptString",
+      "id": "github_token",
+      "description": "GitHub Personal Access Token",
+      "password": true
+    }
+  ],
+  "servers": {
+    "github": {
+      "command": "npx",
+      "args": [ "-y", "@modelcontextprotocol/server-github" ],
+      "env": {
+        "GITHUB_PERSONAL_ACCESS_TOKEN": "GITHUB_PAT"
+      }
+    }
+  }
+}
+```
+
+3. Get a Personal Access Token for your [GitHub Account](https://github.com/settings/personal-access-tokens)
+4. Launch Visual Studio, open the GitHub Copilot window, then click the dropdown that says `Ask`, then select `Agent`.
+![Copilot Agent Mode Selector](./media/vs-2022/copilot-agent/copilotagent-dropdown.png)
+5. When prompted, paste your personal access token into the dialogue.
+![Personal Access Token Dialogue](./media/vs-2022/copilotagent-mcp/mcp-PAT.png)
+
+6. Select the tools you'd like to use: for example, `list_issues`
+![Personal Access Token Dialogue](./media/vs-2022/copilotagent-mcp/mcp-githubtools.png)
+
+7. Try a sample prompt: `list issues assigned to me on GitHub`
+
+8. Copilot will ask for permission to use a tool made available to it by the MCP server, click "Allow" with the scope you wish to proceed with.
+
+![Personal Access Token Dialogue](./media/vs-2022/copilot-agent/copilotagent-toolapproval.png)
```

Comment:
nit - not that it matters, but this appears to be referencing listing commits as opposed to the flow of the example (list issues)